core/src/main/java/feign/Contract.java

```diff
@@ -84,6 +84,9 @@ protected MethodMetadata parseAndValidateMetadata(Class<?> targetType, Method me
       data.method(method);
       data.returnType(Types.resolve(targetType, targetType, method.getGenericReturnType()));
       data.configKey(Feign.configKey(targetType, method));
+      if (AlwaysEncodeBodyContract.class.isAssignableFrom(this.getClass())) {
```

Comment:
This setting is necessary here to avoid having to override this whole method (a copy plus this setting) under `AlwaysEncodeBodyContract`.